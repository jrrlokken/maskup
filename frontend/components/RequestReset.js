import { useMutation } from '@apollo/client';
import gql from 'graphql-tag';

import useForm from '../lib/useForm';
import DisplayError from './ErrorMessage';
import Form from './styles/Form';
import StyledButton from './styles/StyledButton';

const REQUEST_RESET_MUTATION = gql`
  mutation REQUEST_RESET_MUTATION($email: String!) {
    sendUserPasswordResetLink(email: $email) {
      code
      message
    }
  }
`;

const RequestReset = () => {
  const { inputs, handleChange, clearForm } = useForm({ email: '' });
  const [signup, { data, error, loading }] = useMutation(
    REQUEST_RESET_MUTATION,
    {
      variables: inputs,
    }
  );
  async function handleSubmit(event) {
    event.preventDefault();
    const res = await signup().catch(console.error);
    clearForm();
  }

  return (
    <Form
      method='post'
      onSubmit={handleSubmit}
    >
      <h2>Request password reset</h2>
      <DisplayError error={error} />
      <fieldset>
        {data?.sendUserPasswordResetLink === null && (
          <p>Success! Check your email for a reset link.</p>
        )}

        <label htmlFor='email'>
          Email
          <input
            type='email'
            name='email'
            placeholder='email'
            value={inputs.email}
            onChange={handleChange}
          />
        </label>
        <StyledButton type='submit'>Request Reset</StyledButton>
      </fieldset>
    </Form>
  );
}

export default RequestReset;