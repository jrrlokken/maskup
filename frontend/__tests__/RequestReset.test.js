import { configure, mount } from 'enzyme';
import Adapter from '@hteker/enzyme-adapter-react-17';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from '@apollo/client/testing';
import { act } from 'react-dom/test-utils';
import wait from 'waait';

import RequestReset, { REQUEST_RESET_MUTATION } from '../components/RequestReset';

configure({ adapter: new Adapter() });

const resetToken = 'asdfkjksladfjaksdfj';

const mocks = [
  {
    request: {
      query: REQUEST_RESET_MUTATION,
      variables: { email: 'joshualokken@pm.me' },
    },
    result: {
      data: { 
        sendUserPasswordResetLink: null
      },
    }
  }
];

describe('<RequestReset/>', () => {
  it('renders and matches snapshot', async () => {
    const wrapper = mount(
      <MockedProvider>
        <RequestReset />
      </MockedProvider>
    );
    const form = wrapper.find('form[data-test="form"]');
    expect(toJSON(form)).toMatchSnapshot();
  });
  it('calls the mutation', async () => {
    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <RequestReset />
      </MockedProvider>
    );
    wrapper
      .find('input')
      .simulate('change', { target: { name: 'email', value: 'joshualokken@pm.me'} });
    wrapper.find('form').simulate('submit');
    await act(async () => {
      await wait();
      wrapper.update();
      expect(wrapper.find('p').text()).toContain('Success! Check your email for a reset link.');
    });
  });
});
