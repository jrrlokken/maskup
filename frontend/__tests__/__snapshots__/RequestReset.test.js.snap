// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestReset/> renders and matches snapshot 1`] = `null`;
